core/src/main/java/org/wildfly/extras/creaper/core/online/HttpModelControllerClient.java

```diff
@@ -33,87 +34,115 @@
 
     public HttpModelControllerClient(String host, int port, String username, String password) {
         target = new HttpHost(host, port, "http");
-        CredentialsProvider credsProvider = new BasicCredentialsProvider();
+        credsProvider = new BasicCredentialsProvider();
         credsProvider.setCredentials(
                 new AuthScope(target.getHostName(), target.getPort()),
                 new UsernamePasswordCredentials(username, password));
 
-        httpClient = HttpClients.custom()
-                .setDefaultCredentialsProvider(credsProvider)
-                .build();
         url = String.format("http://%s:%s/management", host, port);
     }
 
     private void initialize() {
+        httpClient = HttpClients.custom()
+                .setDefaultRequestConfig(RequestConfig
+                        .copy(RequestConfig.DEFAULT)
+                        .setConnectTimeout(1000)
+                        .setSocketTimeout(1000)
+                        .build())
+                .setDefaultCredentialsProvider(credsProvider)
+                .build();
         localContext = HttpClientContext.create();
     }
 
     @Override
     public ModelNode execute(ModelNode modelNode) throws IOException {
         initialize();
+        ModelNode result;
         try {
             AuthCache authCache = new BasicAuthCache();
             authCache.put(target, getDigestScheme());
             localContext.setAuthCache(authCache);
             CloseableHttpResponse response = httpClient.execute(target, buildRequest(modelNode), localContext);
-            ModelNode m = ModelNode.fromJSONStream(response.getEntity().getContent());
-            response.close();
-            return m;
+            result = ModelNode.fromJSONStream(response.getEntity().getContent());
 
         } catch (MalformedChallengeException e) {
-            e.printStackTrace();
+            throw new IOException(e);
+        } finally {
+            httpClient.close();
         }
-        return null;
+        return result;
     }
 
     @Override
     public ModelNode execute(Operation operation) throws IOException {
+        if (!operation.getInputStreams().isEmpty()) {
+            throw new IllegalStateException("Operation has one or more attachments which is not allowed.");
+        }
         return execute(operation.getOperation());
     }
 
+    /**
+     * Method call normal execution ignoring {@code handler}
+     */
     @Override
     public ModelNode execute(ModelNode modelNode, OperationMessageHandler handler) throws IOException {
         return execute(modelNode);
     }
 
+    /**
+     * Method call normal execution ignoring {@code handler}
+     */
     @Override
     public ModelNode execute(Operation operation, OperationMessageHandler handler) throws IOException {
-        return execute(operation.getOperation());
+        return execute(operation);
     }
 
+    /**
+     * Method call normal execution ignoring {@code handler}
+     */
```

Comment:
1. Not sure what the text means, 2. noone will read the javadoc of this class, it's not a public API. All the restrictions must be documented on the place where the user chooses HTTP. Which currently is `ManagementProtocol.HTTP`, I think.